sunpy/image/transform.py

```diff
@@ -167,7 +101,176 @@ def _get_transform_method(method, use_scipy):
         try:
             import skimage  # NoQA
         except ImportError:
-            warn_user("scikit-image could not be imported. Image rotation will use scipy.")
-            method = 'scipy'
+            raise ImportError("scikit-image must be installed to be usable for rotation.")
 
     return method
+
+
+def add_rotation_function(name, *, handles_clipping=False,
+                          handles_image_nans=False, handles_nan_missing=False):
+    """
+    Decorator to add a rotation function to the library of selectable
+    implementations.
+
+    The decorator accepts the parameters listed under ``Parameters``.  The decorated
+    function must accept the parameters listed under ``Other Parameters`` in that
+    order and return the rotated image.
+
+    Parameters
+    ----------
+    name : `str`
+        The name that will be used to select the rotation function
+    handles_clipping : `bool`
+        Specifies whether the rotation function can internally perform clipping
+    handles_image_nans : `bool`
+        Specifies whether the rotation function can internally handle NaNs in the
+        input image
+    handles_nan_missing : `bool`
+        Specifies whether the rotation function can internally handle NaN as the
+        missing value
+
+    Other Parameters
+    ----------------
+    image : `numpy.ndarray`
+        The image, which could be integers or floats
+    matrix : `numpy.ndarray` that is 2x2
+        The linear transformation matrix (e.g., rotation+scale+skew)
+    shift : 2-element `numpy.ndarray`
+        The translational shift to apply to the image in each axis
+    order : `int`
+        The numerical parameter that controls the degree of interpolation
+    missing : `float`
+        The value to use for outside the bounds of the original image
+    clip : `bool`
+        Whether to clip the output image to the range of the input image
+
+    Notes
+    -----
+    The rotation function is supplied the input image directly, so the function
+    should not modify the image in place.
+    """
+    def decorator(rotation_function):
+        @wraps(rotation_function)
+        def wrapper(image, matrix, shift, order, missing, clip):
+            clip_to_use = clip if handles_clipping else False
+
+            # If missing cannot be used directly, use a value in the input range of the image
+            missing_to_use = (missing if handles_nan_missing or not np.isnan(missing)
+                              else np.nanmin(image))
+
+            if not handles_image_nans:
+                isnan = np.isnan(image)
+                has_nans = np.any(isnan)
+            else:
+                has_nans = False
```

Comment:
Apparently this line isn't covered, I thought we had one rotation method which handled NaNs?

Reply:
I believe the current state is that `scikit-image` can handle image NaNs for some interpolation orders, but (unsurprisingly) not the orders where it calls `scipy` code instead of using native code.  As currently designed, the `handles_image_nans` can't be order-dependent, so I set it to `False` for `scikit-image`.